### ISSUE TITLE
Animation loops before finishing when using dynamic keyframes

### ISSUE THREAD

**Issue by DamianPereira** [wellyshen/use-web-animations#448]
# Bug Report

## Describe the Bug

I have tried creating a HOC with an animated box, which has slightly different animations depending on Math.random(), the animation should run once and finish, but for some reason it keeps looping and each box finishes animating after a random amount of iteration.

## How to Reproduce

```
import "./styles.css";
import useWebAnimations from "@wellyshen/use-web-animations";

export default function App() {
  return (
    <div className="container">
      {[...Array(5).keys()].map((key) => (
        <AnimatedBox key={key} />
      ))}
    </div>
  );
}

const AnimatedBox = () => {
  const { ref } = useWebAnimations({
    keyframes: [{ top: "0%" }, { top: Math.random() > 0.5 ? "50%" : "100%" }],
    animationOptions: {
      duration: 1000,
      fill: "forwards"
    }
  });

  return <div className="box" ref={ref} />;
};
```

styles are only for the flex container and to get the boxes to have size, background, and position relative.

## CodeSandbox Link
[Here's a codesandbox with the issue](https://codesandbox.io/s/beautiful-lederberg-psxmq)

## Expected Behavior

I expect the animation to be calculated once the first time the hook runs, then be executed, and finish, instead of looping.

## Your Environment

- Device: Lenovo Ideapad
- OS: Linux
- Browser: Both Chrome and Firefox tested
- Version: 0.7.4

## Additional Information

Placing the arguments inside a ref and using that fixes the looping:
```
const AnimatedBox = () => {
  const options = useRef({
    keyframes: [{ top: "0%" }, { top: Math.random() > 0.5 ? "50%" : "100%" }],
    animationOptions: {
      duration: 1000,
      fill: "forwards"
    }
  });
  const { ref } = useWebAnimations(options.current);

  return <div className="box" ref={ref} />;
};
```
Should the hook do this internally? Or should this workaround be added to the doc?

**Comment by wellyshen** [comment]
@DamianPereira Hmm... interested, I will investigate this issue soon.

**Comment by wellyshen** [comment]
@DamianPereira I have released [v8.0.0](https://github.com/wellyshen/use-web-animations/releases/tag/v0.8.0) that added the new `shouldUpdateAnimation` option for meeting your case.

**Comment by DamianPereira** [comment]
Thank you!
